azurerm/resource_arm_virtual_machine_scale_set.go

```diff
@@ -1331,8 +1331,9 @@ func flattenAzureRmVirtualMachineScaleSetNetworkProfile(profile *compute.Virtual
 						publicIpConfigs := make([]map[string]interface{}, 0, 1)
 						publicIpConfig := make(map[string]interface{})
 						publicIpConfig["name"] = *publicIpInfo.Name
-						publicIpConfig["domain_name_label"] = *publicIpInfo.VirtualMachineScaleSetPublicIPAddressConfigurationProperties.DNSSettings
+						publicIpConfig["domain_name_label"] = *publicIpInfo.VirtualMachineScaleSetPublicIPAddressConfigurationProperties.DNSSettings.DomainNameLabel
```

Comment:
Nice catch!

Reply:
Indeed! We are going to definitely need to add some.. many nil checks to this as the original code was lacking many. My suggestion is something like this:
```go
if ipCfg := properties.PublicIPAddressConfiguration; ipCfg != nil {
						block := make(map[string]interface{})
						if v := ipCfg.Name; v != nil {
							block["name"] = *v
						}

						if ipProperties := ipCfg.VirtualMachineScaleSetPublicIPAddressConfigurationProperties; ipProperties != nil {
							if dns := ipProperties.DNSSettings; dns != nil {
								if v := dns.DomainNameLabel; v != nil {
									block["domain_name_label"] = *v
								}
							}
							if v := ipProperties.IdleTimeoutInMinutes; v != nil {
								block["idle_timeout"] = *v
							}
						}
						config["public_ip_address_configuration"] = []map[string]interface{}{block}
					}
```